crates/types/src/retries.rs

```diff
@@ -332,6 +332,24 @@ pub fn with_jitter(duration: Duration, max_multiplier: f32) -> Duration {
 
 impl ExactSizeIterator for RetryIter<'_> {}
 
+#[derive(
+    Debug, Copy, Clone, derive_more::Into, derive_more::From, serde::Serialize, serde::Deserialize,
+)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct ExponentiationFactor(f32);
+
+impl PartialEq for ExponentiationFactor {
+    fn eq(&self, other: &Self) -> bool {
+        if self.0.is_nan() && other.0.is_nan() {
+            true
+        } else {
+            self.0 == other.0
+        }
+    }
+}
+
+impl Eq for ExponentiationFactor {}
```

Comment:
I think this can be added to the `derive()` since you already manually implemented the PartialEq trait

Reply:
I tried that, and rustc shouted at me but I didn't look too closely at the reason. I'll try to understand why because it seemed like this should be derivable!